airbyte-integrations/connectors/source-recharge/source_recharge/source.py

```diff
@@ -21,10 +21,12 @@ def get_auth_header(self) -> Mapping[str, Any]:
 
 class SourceRecharge(AbstractSource):
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        auth = RechargeTokenAuthenticator(token=config["access_token"])
+        stream = Shop(authenticator=auth)
         try:
-            auth = RechargeTokenAuthenticator(token=config["access_token"])
-            list(Shop(authenticator=auth).read_records(SyncMode.full_refresh))
-            return True, None
+            result = list(stream.read_records(SyncMode.full_refresh))[0]
```

Comment:
```suggestion
            result = next(stream.read_records(SyncMode.full_refresh))
```

Reply:
Good point, because the `Shop` stream returns a list with the single element) `next` is even better here. Thanks.